src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,230 @@
+}
+
+const getThemeColor = (props, color) => {
+  const {
+    theme: {
+      knitui: { chromaPalette },
+    },
+  } = props
+  return chromaPalette[color].hex()
+}
+
+const getBoxShadowStyle = props => {
+  const {
+    theme: {
+      knitui: { shades },
+    },
+    customProps: { disabled },
+  } = props
+
+  return disabled ? "none" : `0px 0px 2px ${shades.blue50.hex()}`
+}
+
+const getFontSize = props => {
+  const {
+    customProps: { size },
+  } = props
+  if (size.includes("px")) {
+    const avatarSize = Number(size.split("px")[0])
+    return (0.14 * avatarSize) / 2.4
+  } else {
+    return 1.4
+  }
+}
+
+const getLineHeight = props => {
+  const {
+    customProps: { size },
+  } = props
+  if (size.includes("px")) {
+    const avatarSize = Number(size.split("px")[0])
+    return (0.2 * avatarSize) / 2.4
+  } else {
+    return 2
+  }
+}
+
+const getBackgroundColor = (props, state) => {
+  const { isPicture, isName, isIcon, disabled } = props.customProps
+  if (isPicture || isIcon) {
+    return "transparent"
+  } else if (isName) {
+    if (disabled || state === "default") {
+      return getThemeColor(props, "Neutral10")
+    } else if (state === "hover" || state === "focus") {
+      return getThemeColor(props, "Neutral20")
+    } else {
+      return getThemeColor(props, "Neutral10")
+    }
+  }
+}
+
+const getBorderStyle = (props, state) => {
+  const { isPicture, isName, disabled } = props.customProps
+
+  const focusColor = getThemeColor(props, "Azure80")
+  const hoverColor = getThemeColor(props, "Neutral20")
+  const defaultColor = getThemeColor(props, "Neutral10")
+
+  if (state === "focus" && !disabled) {
+    return `1px solid ${focusColor}`
+  }
+  if (isPicture) {
+    if (state === "default" || disabled) {
+      return `1px solid ${defaultColor}`
+    } else if (state === "hover") {
+      return `1px solid ${hoverColor}`
+    }
+  } else if (isName) {
+    return "none"
+  }
+  /* if(isIcon) with (state === "default" || state === "hover" || disabled), focus is already taken care of,
+   * so directly returning transparent style in all other case of isIcon
+   * because of the ts-lint error `not all paths return` to make this condition default
+   */
+  return `1px solid transparent`
+}
+
+const getMarginStyle = props => {
+  const { isPicture, isIcon } = props.customProps
+  return isIcon || isPicture ? "-1px" : "0"
+}
+
+const getFillColor = (props, state) => {
+  const { isPicture, isName, isIcon, disabled } = props.customProps
+
+  if (isPicture || isName) {
+    return "transparent"
+  } else if (isIcon) {
+    if (state === "default" || disabled) {
+      return getThemeColor(props, "Neutral10")
+    } else if (state === "hover" || state === "focus") {
+      return getThemeColor(props, "Neutral20")
+    }
+  }
+}
+
+const getOpacity = props => {
+  const { isIcon } = props.customProps
+
+  return isIcon ? 1 : 0.4
+}
+
+const AvatarBase = styled.button<IStyled<AvatarBaseProps>>`
+  box-sizing: border-box;
```

Comment:
Is this really required? If I remember we had put this in out global styles for all components...